src/pow.cpp

```diff
@@ -43,23 +43,36 @@ unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHead
 
     // Find the first block in the averaging interval
     const CBlockIndex* pindexFirst = pindexLast;
+    arith_uint256 bnAvg;
+    arith_uint256 bnTmp;
     for (int i = 0; pindexFirst && i < params.nPowAveragingWindow; i++) {
+        if (i == 0) {
+            bnAvg.SetCompact(pindexFirst->nBits);
+        } else {
+            bnTmp.SetCompact(pindexFirst->nBits);
+            bnAvg = ((bnAvg * i) + bnTmp) / (i + 1);
```

Comment:
It would make more sense to me to keep just the running total and divide at the end (by declaring ``i`` outside the loop). Maybe I'm microoptimizing.